app/bundles/LeadBundle/Templating/Helper/DncReasonHelper.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * @copyright   2019 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Templating\Helper;
+
+use Mautic\LeadBundle\Entity\DoNotContact;
+use Symfony\Component\Templating\Helper\Helper;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Convert DNC reason ID to text.
+ */
+class DncReasonHelper extends Helper
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * Convert DNC reason ID to text.
+     *
+     * @param int $reasonId
+     *
+     * @return string
+     */
+    public function toText($reasonId)
+    {
+        switch ($reasonId) {
+            case DoNotContact::UNSUBSCRIBED:
+                $reasonKey = 'mautic.lead.event.donotcontact_unsubscribed';
+                break;
+            case DoNotContact::BOUNCED:
+                $reasonKey = 'mautic.lead.event.donotcontact_bounced';
+                break;
+            case DoNotContact::MANUAL:
+                $reasonKey = 'mautic.lead.event.donotcontact_manual';
+                break;
+            default:
+                throw new \InvalidArgumentException(
```

Comment:
Should this throw an exception or return an acceptable string so that other bundles/plugins can extend the reasons for DNC (such as part of a litigators blacklist etc)

Reply:
Also, the default reason is 0 which isnt in this switch's case list so the default behavior will generate an exception.

Reply:
I used new exception class.

TY!